amqp/src/main/scala/akka/stream/alpakka/amqp/impl/AmqpFlow.scala

```diff
@@ -0,0 +1,179 @@
+/*
+ * Copyright (C) 2016-2019 Lightbend Inc. <http://www.lightbend.com>
+ */
+
+package akka.stream.alpakka.amqp.impl
+
+import akka.Done
+import akka.annotation.InternalApi
+import akka.event.Logging
+import akka.stream._
+import akka.stream.alpakka.amqp.{AmqpWriteSettings, WriteMessage, WriteResult}
+import akka.stream.stage.{GraphStageLogic, GraphStageWithMaterializedValue, InHandler, OutHandler, StageLogging}
+
+import scala.concurrent.duration.FiniteDuration
+import scala.concurrent.{Future, Promise}
+import scala.util.{Failure, Success, Try}
+
+/**
+ * Internal API.
+ *
+ * Base stage for AMQP flows that don't use asynchronous confirmations.
+ */
+@InternalApi private sealed abstract class AbstractAmqpFlowStage[T](
+    override val settings: AmqpWriteSettings,
+    promise: Promise[Done],
```

Comment:
I know it wasn't you, but wouldn't `streamCompletion` be a better name?

Reply:
Yup, good catch. Changed.